### DISCUSSION TITLE
Getting rid of the cardano-wallet dependency

### DISCUSSION THREAD

**Discussion by koslambrou** [mlabs-haskell/plutip#193] [Ideas]
The `cardano-wallet` dependency brings delays when wanting to update to the latest cardano-node version. Thus, my question is: can you get rid of `cardano-wallet` completely?

Looking very quickly at the codebase, it seems like you're only using general datatypes declared in `cardano-wallet`. I think these can get replaced pretty easily by types from `cardano-api`. Thoughts?

**Comment by euonymos** [comment]
Hey @koslambrou. This is the really problematic part of the project which hinders maintenance and further development. The reason for that is not only `cardano-wallet` being a dependency but rather the fact that Plutip uses modified local cluster module(s) to allow altering slot length and epoch size as described [here](https://github.com/mlabs-haskell/plutip/blob/master/docs/cardano-wallet-update.md).

So we need a replacement for this part in case we decide to stop using `cardano-wallet`. In fact, we have to options here:
* treat this code as owned and drop the rest of the `cardano-wallet` dependencies
* switch to another (better) cluster launcher that can edit genesis file, I guess `cardano-testnet` from `cardano-node`.
